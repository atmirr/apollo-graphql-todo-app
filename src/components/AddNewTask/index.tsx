import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { TextField } from '@material-ui/core';
import { useMutation } from '@apollo/client';
import CREATE_TASK, { updateCache } from 'src/graphql/mutations/createTask';

const useStyles = makeStyles(() => ({
  input: {
    width: '100%',
    cursor: 'text',
    '& input': {
      height: 25,
    },
  },
}));

function AddNewTask() {
  const classes = useStyles();
  const [createTask] = useMutation(CREATE_TASK, {
    update(cache, { data: { insert_tasks_one: createdTask } }) {
      updateCache(cache, createdTask);
    },
  });
  const handleKeyDown = (e: React.KeyboardEvent) => {
    const element = e.target as HTMLInputElement;
    const title = element.value;
    if (e.keyCode == 13 && title !== '') {
      createTask({ variables: { title } });
      element.value = '';
    }
  };
  return (
    <TextField
      placeholder="Enter a task and press Return ⏎"
      variant="outlined"
      onKeyDown={handleKeyDown}
      className={classes.input}
      autoFocus
    />
  );
}

export default AddNewTask;
