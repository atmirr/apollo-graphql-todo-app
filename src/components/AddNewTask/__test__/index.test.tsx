import React from 'react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { MockedProvider } from '@apollo/client/testing';
import { render, fireEvent } from '@testing-library/react';
import AddNewTask from '../index';

describe('<AddNewTask />', () => {
  it('should renders the component', () => {
    const { container } = render(
      <MockedProvider>
        <AddNewTask />
      </MockedProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('typing in the text field and press enter should make the input empty', () => {
    const { container } = render(
      <MockedProvider>
        <AddNewTask />
      </MockedProvider>,
    );
    const sampleTask = 'New task!';
    const inputNode = container.querySelector('input');
    fireEvent.change(inputNode, { target: { value: sampleTask } });
    expect(inputNode.value).toBe(sampleTask);
    fireEvent.keyDown(inputNode, {
      key: 'Enter',
      code: 'Enter',
      keyCode: 13,
      charCode: 13,
    });
    expect(inputNode.value).toBe('');
  });
});
