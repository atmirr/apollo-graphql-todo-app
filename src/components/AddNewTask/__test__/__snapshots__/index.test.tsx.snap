// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddNewTask /> should renders the component 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root makeStyles-input-1"
>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root Mui-focused Mui-focused MuiInputBase-formControl"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiOutlinedInput-input"
      placeholder="Enter a task and press Return ⏎"
      type="text"
      value=""
    />
    <fieldset
      aria-hidden="true"
      class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      style="padding-left: 8px;"
    >
      <legend
        class="PrivateNotchedOutline-legend-3"
        style="width: 0.01px;"
      >
        <span>
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
